spec/01-unit/01-db/01-schema/11-declarative_config/01-validate_spec.lua

```diff
@@ -908,4 +908,10 @@ describe("declarative config: validate", function()
 
     assert(declarative_config.load(plugins_set))
   end)
+
+  it("rejects plugin with custom DAO in old, hash-like format", function()
+       daos["dao-keywords"] = daos[1]
+       table.remove(daos, 1)
+       assert(not declarative_config.load(plugins_set))
```

Comment:
A simple `assert` is ok for testing the happy path, but we should also test that the error we introduced is the one actually being returned.

```suggestion
       local ok, err = declarative_config.load(plugins_set)
       assert.falsy(ok)
       assert(<something about err>)
```